### PR TITLE
RFD 108 - Agent Census (update)

### PR DESCRIPTION
**Pull request by vitorenesduarte**
This PR marks the RFD as implemented (with https://github.com/gravitational/teleport/pull/22068 and https://github.com/gravitational/cloud/pull/3554).
It also updates the RFD in order to better reflect what was implemented.

[Rendered](https://github.com/gravitational/teleport/blob/vitor/rfd-108-update/rfd/0108-agent-census.md)

### CODE REVIEW

File: rfd/0108-agent-census.md [hunk 1]
```diff
@@ -185,17 +155,22 @@ Following this approach is more reliable than using `/usr/bin/lsb_release` direc
...
 
-On `linux` and `darwin`, `UpstreamInventoryAgentMetadata.HostArchitecture` will be set to the outcome of `arch` (e.g. "x86_64" or "arm64").
-
-The architecture can be different from the `GOARCH` environment variable (e.g. if running the amd64 build of Teleport on an ARM Mac).
-For this reason, we'll use the [`arch`](https://man7.org/linux/man-pages/man1/arch.1.html) command-line utility.
+`UpstreamInventoryAgentMetadata.HostArchitecture` will be set to the value on the `GOARCH` environment variable.
 
-We could also use `uname` but the flag to retrieve the architecture is not portable (`-i` on `linux` and `-p` on `darwin`).
-However, `gopsutil` is using [`golang.org/x/sys/unix.Uname`](https://pkg.go.dev/golang.org/x/sys/unix#Uname) for both OS ([here](https://github.com/shirou/gopsutil/blob/v3.23.1/host/host_posix.go)), so we may consider to do the same.
+In the future we may use `sysctl -n sysctl.proc_translated` in order to detect if the agent is running under Rosetta.
```
Comment (espadolini) [inline rfd/0108-agent-census.md:160]:
```suggestion
In the future we may use `sysctl -n sysctl.proc_translated` in order to detect if a macOS agent is running under Rosetta.
```

**Review by fspmarshall — APPROVED** [review]

File: rfd/0108-agent-census.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 authors: Vitor Enes (vitor@goteleport.com)
-state: draft
+state: implemented
```
Comment (zmb3) [inline rfd/0108-agent-census.md:3]: If this has been released, can you also mark the release it was introduced in?

Comment (vitorenesduarte) [reply rfd/0108-agent-census.md:3]: The teleport backports have not been released yet. I'll open a follow-up PR once they are.

Comment (zmb3) [reply rfd/0108-agent-census.md:3]: If they're already merged, you can just assume they'll be in the next release too.

Comment (vitorenesduarte) [reply rfd/0108-agent-census.md]: Done in a4292a59c107183fba183e0ab8adb850c4aea7be. It assumes that the next releases for v11 and v12 will be patch releases.

**Review by espadolini — APPROVED** [review]

**Pull request merged**